### PR TITLE
IOS: File Protection Options

### PR DESCRIPTION
**Pull request by gensc004**
Half of solution for [HouseRater/issues/3302](https://github.com/houserater/HouseRater/issues/3302)

- Added support for specifying file protections when creating a directory or moving, copying, and creating a file

### CODE REVIEW

File: RNFSManager.m [hunk 1]
```diff
@@ -868,7 +903,13 @@ - (NSDictionary *)constantsToExport
            @"RNFSTemporaryDirectoryPath": NSTemporaryDirectory(),
            @"RNFSLibraryDirectoryPath": [self getPathForDirectory:NSLibraryDirectory],
            @"RNFSFileTypeRegular": NSFileTypeRegular,
-           @"RNFSFileTypeDirectory": NSFileTypeDirectory
+           @"RNFSFileTypeDirectory": NSFileTypeDirectory,
+           @"RNFSFileProtectionKeys": @{
+               @"NSFileProtectionCompleteUntilFirstUserAuthentication":NSFileProtectionCompleteUntilFirstUserAuthentication,
```
Comment (hnryjms) [inline RNFSManager.m]: I think w eshould maybe use `RCT_ENUM_CONVERTER` here... check out [this PR](https://github.com/facebook/react-native/pull/14102) I made into React Native for modals. Basically, it adds the ability to cast a `String *` into `NSFileProtection`

File: FS.common.js [hunk 2]
```diff
@@ -29,6 +29,11 @@ var normalizeFilePath = (path: string) => (path.startsWith('file://') ? path.sli
 
 type MkdirOptions = {
   NSURLIsExcludedFromBackupKey?: boolean; // iOS only
+  NSFileProtectionKey?: string; // IOS only
+};
+
+type FileOptions = {
+    NSFileProtectionKey?: string; // IOS only
```
Comment (hnryjms) [inline FS.common.js]: Once we have `RCT_ENUM_CONVERTER`, this will be an enum of strings.

File: RNFSManager.m (hunk 1 above)
Comment (hnryjms) [reply RNFSManager.m]: I'm mainly worried that Apple generally has their enums map to not-serializable values internally most of the time.

File: FS.common.js [hunk 3]
```diff
@@ -357,7 +362,7 @@ var RNFS = {
     return RNFSManager.copyAssetsVideoIOS(imageUri, destPath);
   },
 
-  writeFile(filepath: string, contents: string, encodingOrOptions?: any): Promise<void> {
+  writeFile(filepath: string, contents: string, encodingOrOptions?: any, fileOptions: FileOptions = {}): Promise<void> {
```
[the remark this answers is not in the document]
Comment (gensc004) [reply FS.common.js]: Yeah I thought about using it, but I decided to stay away from it cause it was meh

[the remark this answers is not in the document]
Comment (hnryjms) [reply FS.common.js]: I don't think that justifies this...

**Pull request merged**